packages/nest/package.py

```diff
@@ -0,0 +1,179 @@
+                    c_compiler = self.spec['mpi'].mpicc
+                    cxx_compiler = self.spec['mpi'].mpicxx
+
+            cmake_options = ['-DCMAKE_INSTALL_PREFIX:PATH=%s' % prefix,
+                             '-DCMAKE_C_FLAGS=%s' % optflag,
+                             '-DCMAKE_CXX_FLAGS=%s' % optflag,
+                             '-DCMAKE_C_COMPILER=%s' % c_compiler,
+                             '-DCMAKE_CXX_COMPILER=%s' % cxx_compiler,
+                             '-Dwith-ltdl=OFF',
+                             '-Dwith-readline=OFF']
+
+            # not sure the -Dwith-optimize=flags option is required if we already set CMAKE_C_FLAGS above
+            # but just to be safe
+            semicolon_separated_optflag = optflag.translate(string.maketrans(' ', ';'))
+            cmake_options.extend(['-Dwith-optimize=' + semicolon_separated_optflag])
+
+            if self.spec.satisfies('~python'):
+                cmake_options.extend(['-Dwith-python=OFF'])
+                cmake_options.extend(['-Dcythonize-pynest=OFF'])
+            else:
+                cmake_options.extend(['-Dwith-python=ON'])
+                cmake_options.extend(['-Dcythonize-pynest=%s' % self.spec['py-cython'].prefix])
+
+            if self.spec.satisfies('~gsl'):
+                cmake_options.extend(['-Dwith-gsl=OFF'])
+            else:
+                cmake_options.extend(['-Dwith-gsl=%s' % self.spec['gsl'].prefix])
+
+            if self.spec.satisfies('~mpi'):
+                cmake_options.extend(['-Dwith-mpi=OFF'])
+            else:
+                cmake_options.extend(['-Dwith-mpi=ON'])
+
+            if self.spec.satisfies('~openmp'):
+                cmake_options.extend(['-Dwith-openmp=OFF'])
+            else:
+                cmake_options.extend(['-Dwith-openmp=ON'])
+
+            if self.spec.satisfies('~static'):
+                cmake_options.extend(['-Dstatic-libraries=OFF'])
+            else:
+                cmake_options.extend(['-Dstatic-libraries=ON'])
+
+            if 'cray' in self.spec.architecture:
+                cmake_options.extend(['-Dwith-warning=OFF'])
+
+            cmake('..', *cmake_options)
+            self.profiling_wrapper_on()
+            make()
+            make('install')
+            self.profiling_wrapper_off()
+
+    def setup_environment(self, spack_env, run_env):
+        exe = '%s/nest' % self.prefix.bin
+        run_env.set('NEST_EXE', exe)
+
+        # for mvapich2 module we need to setup MV2_ENABLE_AFFINITY
+        # env variable to turn on multi-threading support
+        run_env.set('MV2_ENABLE_AFFINITY', '0')
+
+        # I wish there was a way to detect the specs here.
+        # If the user didn't want python, no need to pollute their env.
+        # How to get python version specified by user???
+        if self.spec.satisfies('+python'):
+            py_version_string = 'python{0}'.format(self.spec['python'].version.up_to(2))
+            try:
```

Comment:
there is another easy/lightweight way to achieve the same : we can use [llnl.util.filesystem](http://spack.readthedocs.io/en/latest/llnl.util.html?highlight=find#module-llnl.util.filesystem) to find `PyNEST egg` file and get `dirname`.